### ISSUE TITLE
study_citation and study_source_citation columns contain no data

### ISSUE THREAD

**Issue by ZekeMarshall** [ropensci/rglobi#38]
I was wondering whether it would be possible to include data for the source of the observed interaction when using `rglobi::get_interactions()` to retrieve data. At the moment the `study_citation` and `study_source_citation` columns contain only `NA` values.

Thanks for your work on the package, it's a fantastic resource.

**Comment by jhpoelen** [comment]
hey @ZekeMarshall thanks for your message.

To include the study_ciration and study_source_citation, you'll have to retrieve individual indexed observations, not just indexed taxon-taxon interactions using:

e.g.,

```R
> head(rglobi::get_interactions(taxon='Enhydra lutris', returnobservations=T))
  source_taxon_external_id      source_taxon_name                             
1        INAT_TAXON:133061 Enhydra lutris kenyoni
2        INAT_TAXON:117520  Enhydra lutris nereis
3        INAT_TAXON:117520  Enhydra lutris nereis
4        INAT_TAXON:117520  Enhydra lutris nereis
5        INAT_TAXON:117520  Enhydra lutris nereis
6        INAT_TAXON:117520  Enhydra lutris nereis
                                                                           source_taxon_path
1 Animalia | Chordata | Mammalia | Carnivora | Mustelidae | Enhydra | Enhydra lutris kenyoni
2  Animalia | Chordata | Mammalia | Carnivora | Mustelidae | Enhydra | Enhydra lutris nereis
3  Animalia | Chordata | Mammalia | Carnivora | Mustelidae | Enhydra | Enhydra lutris nereis
4  Animalia | Chordata | Mammalia | Carnivora | Mustelidae | Enhydra | Enhydra lutris nereis
5  Animalia | Chordata | Mammalia | Carnivora | Mustelidae | Enhydra | Enhydra lutris nereis
6  Animalia | Chordata | Mammalia | Carnivora | Mustelidae | Enhydra | Enhydra lutris nereis
  source_specimen_life_stage interaction_type target_taxon_external_id
1                       <NA>          preysOn             GBIF:2289226
2                       <NA>          preysOn         INAT_TAXON:47828
3                       <NA>          preysOn         INAT_TAXON:47828
4                       <NA>          preysOn             GBIF:2285679
5                       <NA>          preysOn                 GBIF:137
6                       <NA>          preysOn         INAT_TAXON:47826
  target_taxon_name
1           Octopus
2         Cancridae
3         Cancridae
4           Mytilus
5          Bivalvia
6  Cancer productus
                                                                                                                           target_taxon_path
1                                                                       Animalia | Mollusca | Cephalopoda | Octopoda | Octopodidae | Octopus
2 Animalia | Arthropoda | Crustacea | Multicrustacea | Malacostraca | Eucarida | Decapoda | Pleocyemata | Brachyura | Cancroidea | Cancridae
3 Animalia | Arthropoda | Crustacea | Multicrustacea | Malacostraca | Eucarida | Decapoda | Pleocyemata | Brachyura | Cancroidea | Cancridae
4                                                                            Animalia | Mollusca | Bivalvia | Mytilida | Mytilidae | Mytilus
5                                                                                                             Animalia | Mollusca | Bivalvia
6                                                    Animalia | Arthropoda | Malacostraca | Decapoda | Cancridae | Cancer | Cancer productus
  target_specimen_life_stage latitude longitude
1                       <NA> 56.93835 -135.7816
2                       <NA> 35.39714 -120.9534
3                       <NA> 35.39714 -120.9534
4                       <NA> 36.95229 -121.6419
5                       <NA> 36.90766 -121.7446
6                       <NA> 36.99515 -121.6459
                                                                                                                                                       study_citation
1        Donna Pomeroy. 2012. Enhydra lutris kenyoni prey id Octopus. iNaturalist.org. Accessed at <https://www.inaturalist.org/observations/1181454> on 25 Jul 2021.
2      Marisa Agarwal. 2014. Enhydra lutris nereis prey id Cancridae. iNaturalist.org. Accessed at <https://www.inaturalist.org/observations/1130763> on 25 Jul 2021.
3      Marisa Agarwal. 2014. Enhydra lutris nereis prey id Cancridae. iNaturalist.org. Accessed at <https://www.inaturalist.org/observations/1130763> on 25 Jul 2021.
4          James Maughn. 2015. Enhydra lutris nereis prey id Mytilus. iNaturalist.org. Accessed at <https://www.inaturalist.org/observations/1964813> on 25 Jul 2021.
5         James Maughn. 2014. Enhydra lutris nereis prey id Bivalvia. iNaturalist.org. Accessed at <https://www.inaturalist.org/observations/1119439> on 25 Jul 2021.
6 Donna Pomeroy. 2014. Enhydra lutris nereis prey id Cancer productus. iNaturalist.org. Accessed at <https://www.inaturalist.org/observations/799531> on 25 Jul 2021.
                                                                                                                        study_source_citation
1 http://iNaturalist.org is a place where you can record what you see in nature, meet other nature lovers, and learn about the natural world.
2 http://iNaturalist.org is a place where you can record what you see in nature, meet other nature lovers, and learn about the natural world.
3 http://iNaturalist.org is a place where you can record what you see in nature, meet other nature lovers, and learn about the natural world.
4 http://iNaturalist.org is a place where you can record what you see in nature, meet other nature lovers, and learn about the natural world.
5 http://iNaturalist.org is a place where you can record what you see in nature, meet other nature lovers, and learn about the natural world.
6 http://iNaturalist.org is a place where you can record what you see in nature, meet other nature lovers, and learn about the natural world.
```

Please let me know if this works for you. 

Also, please suggest any changes you'd like to see to documentation to help others (or your past/future self) make this easier to find. 


returnobservations = T

**Comment by ZekeMarshall** [comment]
Hi @jhpoelen,

Thank you very much for your quick and comprehensive response.

Using `returnobservations = T` when calling the rglobi::get_interactions() function works perfectly.

I had a look through the documentation and found the  `returnobservations = T` parameter in the `rglobi::get_interactions_by_taxa` function, however there is no reference to it in the "Introduction to the rglobi package" article, so it might be helpful to include it there.

Thanks again!

Zeke

**Comment by jhpoelen** [comment]
As suggested @ZekeMarshall , a section was added to the vignette to give an example of ```returnobservations=T```. See https://github.com/ropensci/rglobi/issues/38#issuecomment-897603331 . Open to suggestions!

**Comment by jhpoelen** [comment]
Thanks to @ZekeMarshall and CRAN's excellent release system, the just released rglobi v0.2.25 ```introduction``` vignette now includes a section "Find citations using individual interaction records" with examples for returnobservations=T . 

Happy to accept pull requests with additional improvements to the documentation (and code . . . ). 

See attached screenshot captured from https://cran.r-project.org/web/packages/rglobi/vignettes/rglobi_vignette.html at 12 Aug 2021 . . 

![Screenshot from 2021-08-12 10-02-45](https://user-images.githubusercontent.com/1084872/129238426-273d73b4-dd36-4332-9a34-a98edb46712b.png)
